### DISCUSSION TITLE
error trying to import .tif

### DISCUSSION THREAD

**Discussion by Jeatable** [AICAN-Research/FAST-Pathology#99] [Q&A] [answered]
When I try to import tif I get "ERROR [26500] FAST exception caught in Qt event handler OpenSlide was unable to open the file:...". The images from FastPathology Test project open without any error. My files are about 180 MB, and names are pretty long with dashes and underscores, so I tried renaming my tif to A01.tif and A01.tiff, and tried random 27 kB tif,  resulting in the same error. I was able to process my tif with long filename in QuPath and MIB. Am I missing something, or should I post as issue?

**Comment by andreped** [comment, marked as answer]
Hello, @Jeatable! :]

Hard for me to say what the problem is within more details into what image you are attempting to use.

FastPathology only supports pyramidal images. If you attempt to open a regular non-pyramidal TIFF, it will not be able to render it. FastPathology was developed to render WSIs, not smaller regions only which may be stored as non-pyramidal image formats (e.g., PNG).

The reason why QuPath opens non-pyramidal images, is because it pyramidizes the image, if it is not already pyramidal - this can be very slow for larger WSIs.

I would assume you are trying to visualize non-pyramidal TIFFs?

**Reply by Jeatable** [reply]
Hi @andreped,

Thank you for your quick response!

Yes you assume correctly, that is exactly what I'm trying. So my workaround would be getting my TIFFs converted to pyramidal. Or rather after I stitched them together in imageJ save them as pyramidal TIFF. Otherwise I suppose I could also just export from QuPath?

And consequently, would compression for saving as pyramidal TIFF influence the effectivity of any model I would want to try in FastPathology? 

Update: After using Kheops plugin to save as pyramidal OME-TIFF, I was able to import my files in FastPathology. However I'm using raw images and the thumbnail is black, which is expected behavior I guess since most 'normal' image viewers besides imageJ do this (not auto scaling pixel values or something), but the viewer pane remains empty. And I dont see the button for the .onnx file I placed in the models folder alongside the txt file. Perhaps I should make a new discussion string for this..

**Reply by andreped** [reply]
You could just use the command line tool [vips](https://github.com/libvips/libvips/wiki/Build-for-Ubuntu#installing-the-maintained-ubuntudebian-binary-packages). See link for how to install it on Ubuntu (assuming your are on Ubuntu).

Then to convert, do something like this:
```
vips tiffsave /path/to/original/image.tiff /path/to/converted/pyramidal/image.tiff --bigtiff --tile --pyramid --compression=jpeg --Q=85
```

This example include compression as well. Note that not "any" JPEG compression is supported sadly.

---
EDIT: BTW: Which original format are you converting from? I assume you have some kind of strange WSI format, like Olympus cellSens VSI, or similar?

**Reply by Jeatable** [reply]
Ha, you beat me to it. They are 'raw' TIFF files from an EVOS M7000. I'm on Win 10.

**Reply by andreped** [reply]
The scanner does not produce a pyramidal image format like .vsi or .svs or similar?

**Reply by Jeatable** [reply]
No, it lets you directly save to .tif or .png. Even z-stacks are directly saved to these formats.

**Reply by andreped** [reply]
Oh, that's unfortunate and a bit surprising tbh...

Anyways, to install `vips` on Windows see [here](https://www.libvips.org/install.html).

**Reply by Jeatable** [reply]
Using vips to generate a pyramidal tiff worked, also solved the 'imported but not being able to view' problem. 

Now I'm trying to apply an .onnx model. Initially placed it in the /fastpathology/models folder but didn't find a possiblity to apply it in FASTPathology. 

So I downloaded one of the existing colon HE models, and mimicked its structure. Created a title-model and title-fp folder in datahub folder, placed my .onnx file in the title-model folder and changed PipelineName, Attribute classes and Attribute model in the colon HE pipeline.fpl file and saved it to the new title-fp folder.

Two warnings in fastpahtology but it seems to be running,
WARNING [16048] Requested magnification level does not exist in image pyramid. Will now try to sample from a lower level and resize. This may increase runtime.
WARNING [16048] Patch size must be a multiple of 16 (TIFF limitation). Adding some overlap (32, 32) to fix.

However it has been running for over 3 hours, with the progress bar still at 0%. Is this normal during auto-tuning, or should I edit the .fpl file to prevent the warnings?

**Reply by andreped** [reply]
@Jeatable Could you open a new issue or discussion about this? Makes it easier for me and others to track different threads :]

I think I know what the issue is, but lets take it in a separate issue/discussion.

**Reply by Jeatable** [reply]
Will do! 

Thanks for your help!
